base/compiler/typelimits.jl

```diff
@@ -128,7 +128,15 @@ function _limit_type_size(@nospecialize(t), @nospecialize(c), sources::SimpleVec
         end
         return Vararg{VaT}
     elseif isa(t, DataType)
-        if isa(c, DataType)
+        if isa(c, Core.TypeofVararg)
+            # Tuple{Vararg{T}} --> Tuple{T} is OK
+            return _limit_type_size(t, c.T, sources, depth, 0)
```

Comment:
```suggestion
            return _limit_type_size(t, unwrapva(c), sources, depth, 0)
```
for the consistency with the other parts ?

Reply:
@keno ^